client/me/security-2fa-backup-codes/index.jsx

```diff
@@ -3,61 +3,52 @@
 /**
  * External dependencies
  */
-
 import React from 'react';
-import createReactClass from 'create-react-class';
+import { connect } from 'react-redux';
 import { localize } from 'i18n-calypso';
-import debugFactory from 'debug';
-const debug = debugFactory( 'calypso:me:security:2fa-backup-codes' );
 
 /**
  * Internal dependencies
  */
-import Security2faBackupCodesPrompt from 'me/security-2fa-backup-codes-prompt';
-import SectionHeader from 'components/section-header';
 import Button from 'components/button';
 import Card from 'components/card';
-import eventRecorder from 'me/event-recorder';
-import twoStepAuthorization from 'lib/two-step-authorization';
-import Security2faBackupCodesList from 'me/security-2fa-backup-codes-list';
 import Notice from 'components/notice';
+import SectionHeader from 'components/section-header';
+import Security2faBackupCodesList from 'me/security-2fa-backup-codes-list';
+import Security2faBackupCodesPrompt from 'me/security-2fa-backup-codes-prompt';
+import twoStepAuthorization from 'lib/two-step-authorization';
+import { recordGoogleEvent } from 'state/analytics/actions';
 
-const Security2faBackupCodes = createReactClass( {
-	displayName: 'Security2faBackupCodes',
-
-	mixins: [ eventRecorder ],
-
-	componentDidMount: function() {
-		debug( this.constructor.displayName + ' React component is mounted.' );
-	},
-
-	componentWillUnmount: function() {
-		debug( this.constructor.displayName + ' React component will unmount.' );
-	},
-
-	getInitialState: function() {
-		var printed = this.props.userSettings.getSetting( 'two_step_backup_codes_printed' );
+class Security2faBackupCodes extends React.Component {
+	constructor( props ) {
+		super( props );
+		const printed = this.props.userSettings.getSetting( 'two_step_backup_codes_printed' );
 
-		return {
-			printed: printed,
+		this.state = {
+			printed,
 			verified: printed,
 			showPrompt: ! printed,
 			backupCodes: [],
 			generatingCodes: false,
 		};
-	},
+	}
+
+	handleGenerateButtonClick = () => {
+		this.props.recordGoogleEvent( 'Me', 'Clicked on Generate New Backup Codes Button' );
+		this.onGenerate();
+	};
```

Comment:
I think I'd rather merge this into `onGenerate`, since it's the only instance where it's used.

Reply:
Makes sense, done in 1b6a50f72b59f5c49132c71f4b022c3ee206abfc.